### PR TITLE
moved PathValidator from NuGet.Core to NuGet.Configuration

### LINKED ISSUE #865: Using relative paths for package sources is problematic to use
* NuGet.exe (2.8.6) does not allow specifying a relative path
```
> nuget.exe source -config $PWD\NuGet.config add -Name RelativeSourceName -Source RelativePath
The source specified is invalid. Please provide a valid source.
```

* Relative paths get overwritten by absolute paths when `SavePackageSource` is invoked. Reproes with NuGet 2.8.6 and 3.0.

### PR DESCRIPTION
**Pull request by zhili1208**
Fixed Relative path
fixed https://github.com/NuGet/Home/issues/865

### CODE REVIEW

File: src/NuGet.Core/NuGet.Configuration/Utility/PathValidator.cs [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        /// </remarks>
+        /// <param name="path">The path to validate.</param>
+        /// <returns>True if valid, False if invalid.</returns>
+        [SuppressMessage("Microsoft.Design", "CA1031:DoNotCatchGeneralExceptionTypes", Justification = "We don't want to throw during detection")]
+        public static bool IsValidUncPath(string path)
+        {
+            try
+            {
+                Path.GetFullPath(path);
+                return Regex.IsMatch(path.Trim(), @"^\\\\");
```
Comment (alpaix) [inline src/NuGet.Core/NuGet.Configuration/Utility/PathValidator.cs]: Generally we should avoid using regex's. Especially in this case when 2-chars comparison would be straightforward.

**Comment by alpaix** [comment]
@zhili1208 Please add unit-tests for all valid/invalid cases.

**Comment by emgarten** [comment]
:+1: on tests, they should cover as many scenarios as possible so that we can verify this works on Linux and OSX when the CI runs the tests.

You might also want to take a look at UriUtility which helps with Uri.TryCreate for linux paths.
//cc @spadapet

File: src/NuGet.Core/NuGet.Configuration/Utility/PathValidator.cs [hunk 2]
```diff
@@ -0,0 +1,104 @@
+﻿using System;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.Text.RegularExpressions;
+using NuGet.Common;
+
+namespace NuGet.Configuration
+{
+    public static class PathValidator
```
Comment (alpaix) [inline src/NuGet.Core/NuGet.Configuration/Utility/PathValidator.cs]: I think this utility class should be in `NuGet.Common`. @emgarten ?

Comment (emgarten) [inline src/NuGet.Core/NuGet.Configuration/Utility/PathValidator.cs]: :+1: I agree, NuGet.Common would be best, this seems like a very common operation.

**Comment by emgarten** [comment]
:shipit:

**Pull request closed without merging**